transport-native-epoll/src/main/java/io/netty/channel/epoll/AIOContext.java

```diff
@@ -109,13 +138,40 @@ public void processReady(AIOEpollFileChannel file) {
                     throw new IllegalStateException("" + id + " missing");
                 }
 
+                // Finally complete the request
                 int len = (int) lengthRead;
                 assert len == lengthRead;
                 assert len >= 0 : len;
                 request.buffer.limit(len);
                 request.buffer.position(len);
                 request.handler.completed(len, request.attachment);
             }
+
+            //add any pending reads
+            //We do this after processing the finished reads incase any outstanding reads
+            //they share the same read buffer
+            for (int i = 0; i < numReady; i++) {
+                --outstanding;
+                IORequest req = null;
+                // Push any pending requests if we have room
+                try {
+                    if (outstanding + 1 <= maxOutstanding && (req = pendingRequests.poll()) != null) {
+                        assert req.file.epollEventLoop.aioContext == this;
+                        long nextId = Native.submitAIORead(this, req.file.getEventFd(), req.file.getFd(),
+                                                           req.position, req.length, req.buffer);
+                        IORequest r = outstandingRequests.putIfAbsent(nextId, req);
+                        assert r == null;
```

Comment:
If we wanted to refactor the code a little bit, we should have a dedicated method to submit the AIO read, add the request to `outstandingRequests` and, if required, increment `outstanding`. This functionality is repeated twice.